lib/travis/api/v3/services/env_vars/for_repository.rb

```diff
@@ -0,0 +1,9 @@
+module Travis::API::V3
+  class Services::EnvVars::ForRepository < Service
+    def run!
+      raise LoginRequired unless access_control.logged_in? or access_control.full_access?
+      raise NotFound      unless repository = find(:repository)
```

Comment:
I would actually make this logic a method on service that you can call since it's the same logic 5 times:
here:
``` ruby
def run!
  repository = verify_access_and_get_repository
  ....etc
end
```
on Service:
``` ruby 
def verify_access_and_get_repository
  raise LoginRequired unless access_control.logged_in_or_full_access?
  find(:repository) or raise NotFound 
end
```